internal/scanner/scanner.go

```diff
@@ -0,0 +1,166 @@
+package scanner
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+)
+
+const (
+	Eof = iota
+	Whitespace
+	LeftParen
+	RightParen
+	LeftBracket
+	RightBracket
+	Semicolon
+	Backslash
+	String
+	Illegal
+)
+
+var eof = rune(0)
+
+// isWhiteSpace is a check to see if a rune is whitespace (space, tab, newline, carriage return)
+func isWhitespace(ch rune) bool {
+	return ch == ' ' || ch == '\t' || ch == '\n' || ch == '\r'
+}
+
+// isSpecial is a check for special characters used in SGF files
+func isSpecial(ch rune) bool {
+	return ch == '(' || ch == ')' || ch == '[' || ch == ']' || ch == ';'
+}
+
+// The scanner will "tokenize" the input into Token structs
+type Token struct {
```

Comment:
Nit: Comment should have form "Token does blah"

Golint should catch these sorts of errors